### PR TITLE
DOP-2602: Add option for external links to open in a new tab/window

### PR DESCRIPTION
**Pull request by gjchong25**
### Stories/Links:

[DOP-2602](https://jira.mongodb.org/browse/DOP-2602)

### Current Behavior:

[Atlas Production Example 1](https://www.mongodb.com/docs/atlas/tutorial/create-atlas-account/#google-accounts)
[Atlas Production Example 2](https://www.mongodb.com/docs/atlas/reference/unsupported-commands/#unsupported-commands-in-m0-m2-m5-clusters)

### Staging Links:

[Atlas Staging Example 1](https://docs-mongodbcom-integration.corp.mongodb.com/master/cloud-docs/grace.chong/DOP-2602/tutorial/create-atlas-account/#google-accounts)
[Atlas Staging Example 2](https://docs-mongodbcom-integration.corp.mongodb.com/master/cloud-docs/grace.chong/DOP-2602/reference/unsupported-commands/#unsupported-commands-in-m0-m2-m5-clusters) 

### Notes:

In summary, any link that isn't to `www.mongodb.com/docs/*` should opens in a new tab and should be styled according to LG's new link component found [here](https://www.mongodb.design/component/typography/example/), even if the link is a MDB link to a non-documentation site. Links that are internal MDB docs links should be styled with blue text and no link icon, and should not open in a new tab. 

Right now, since all of the URLs may not have been changed yet after the subdomain consolidation project, the old docs URL format has been included in the logic (this can be removed once all links have been changed to reflect the updated URL pattern by content folks).

### CODE REVIEW

**Review by rayangler — COMMENTED** [review]
Things LGTM for the most part but flagging a couple of concerns for now

File: src/components/Link.js [hunk 1]
```diff
@@ -27,6 +28,8 @@ const Link = ({ children, to, activeClassName, partiallyActive, ...other }) => {
         {children}
       </GatsbyLink>
     );
+  } else if (!anchor && !(to.includes('www.mongodb.com/docs/') || to.match(/docs.*mongodb.com/))) {
+    return <LGLink href={to}>{children}</LGLink>;
```
[the remark this answers is not in the document]
Comment (allisonmui9) [reply src/components/Link.js:32]: Thanks for the callout! Should probably do just one - the LG's light gray underline. I think it'd be more aligned with the brand update later on.

[the remark this answers is not in the document]
Comment (gjchong25) [reply src/components/Link.js:32]: Just to confirm, the non-LG links should remain with the normal blue underline?

[the remark this answers is not in the document]
Comment (rayangler) [reply src/components/Link.js]: I think it makes sense to keep non-LG links as-is for now, and then eventually transition our other links to use LG?

[the remark this answers is not in the document]
Comment (allisonmui9) [reply src/components/Link.js]: yes makes sense! Non-LG links with the normal blue underline and LG external links with the grey underline.
